core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,556 @@
-          _topicIsLoading = false;
-        });
-      },
+    }
+    return null;
+  }
+  _updateClassroomUrlFragment(classroomUrlFragment: string): void {
+    this._classroomUrlFragment = classroomUrlFragment;
+  }
+  _updateTopic(
+      newBackendTopicDict: TopicBackendDict,
+      skillIdToDescriptionDict: SkillIdToDescriptionMap): void {
+    this._setTopic(
+      this.topicObjectFactory.create(
+        newBackendTopicDict, skillIdToDescriptionDict));
+  }
+  private _updateSkillIdToRubricsObject(
+      skillIdToRubricsObject:
+      {[skillId: string]: RubricBackendDict[]}): void {
+    for (let skillId in skillIdToRubricsObject) {
+      let rubrics = skillIdToRubricsObject[skillId].map((
+          rubric) => {
+        return this.rubricObjectFactory.createFromBackendDict(rubric);
+      });
+      this._skillIdToRubricsObject[skillId] = rubrics;
+    }
+  }
+  private _setSubtopicPage(subtopicPage: SubtopicPage): void {
+    this._subtopicPage.copyFromSubtopicPage(subtopicPage);
+    this._cachedSubtopicPages.push(cloneDeep(subtopicPage));
+    this._subtopicPageLoadedEventEmitter.emit();
+  }
+  private _updateSubtopicPage(
+      newBackendSubtopicPageObject: SubtopicPageBackendDict): void {
+    this._setSubtopicPage(this.subtopicPageObjectFactory.createFromBackendDict(
+      newBackendSubtopicPageObject));
+  }
+  private _setTopicRights(topicRights: TopicRights): void {
+    this._topicRights.copyFromTopicRights(topicRights);
+  }
+  private _updateTopicRights(
+      newBackendTopicRightsObject): void {
+    this._setTopicRights(TopicRights.createFromBackendDict(
+      newBackendTopicRightsObject));
+  }
+  private _setCanonicalStorySummaries(
+      canonicalStorySummaries: StorySummaryBackendDict[]): void {
+    this._canonicalStorySummaries = canonicalStorySummaries.map(
+      (storySummaryDict) => {
+        return StorySummary.createFromBackendDict(
+          storySummaryDict);
+      });
+    this._storySummariesInitializedEventEmitter.emit();
+  }
 
-      getGroupedSkillSummaries: function() {
-        return angular.copy(_groupedSkillSummaries);
-      },
+  private _setTopicWithNameExists(topicWithNameExists: boolean): void {
+    this._topicWithNameExists = topicWithNameExists;
+  }
 
-      getSkillQuestionCountDict: function() {
-        return _skillQuestionCountDict;
-      },
+  private _setTopicWithUrlFragmentExists(
+      topicWithUrlFragmentExists: boolean): void {
+    this._topicWithUrlFragmentExists = topicWithUrlFragmentExists;
+  }
 
-      /**
-       * Returns whether the topic name already exists on the server.
-       */
-      getTopicWithNameExists: function() {
-        return _topicWithNameExists;
+  /**
+   * Loads, or reloads, the topic stored by this service given a
+   * specified topic ID. See setTopic() for more information on
+   * additional behavior of this function.
+   */
+  loadTopic(topicId: string): void {
```

Comment:
This does an async operation and therefore we had used a `$rootScope.$applyAsync();`. but that has been removed. How are we handling this now?

Reply:
Made the function to return a promise. Every angularJS file where this function was called, I have added an $rootScope.$applyAsync();. This removes the requirement of using $rootScope.$applyAsync(); in the function itself